frontend/src/component/project/Project/ProjectSettings/ChangeRequestConfiguration/ChangeRequestConfiguration.tsx

```diff
@@ -1,45 +1,45 @@
-import { PageContent } from 'component/common/PageContent/PageContent';
-import { PageHeader } from 'component/common/PageHeader/PageHeader';
-import { TextCell } from 'component/common/Table/cells/TextCell/TextCell';
-import PermissionSwitch from 'component/common/PermissionSwitch/PermissionSwitch';
-import { UPDATE_FEATURE_ENVIRONMENT } from 'component/providers/AccessProvider/permissions';
-import { useRequiredPathParam } from 'hooks/useRequiredPathParam';
-import { Dialogue } from 'component/common/Dialogue/Dialogue';
-import { ConditionallyRender } from 'component/common/ConditionallyRender/ConditionallyRender';
+import { useMemo, useState, VFC } from "react";
+import { HeaderGroup, useGlobalFilter, useTable } from "react-table";
+import { Alert, Box, Typography } from "@mui/material";
+import { SortableTableHeader, Table, TableBody, TableCell, TableRow } from "component/common/Table";
+import { sortTypes } from "utils/sortTypes";
+import { PageContent } from "component/common/PageContent/PageContent";
+import { PageHeader } from "component/common/PageHeader/PageHeader";
+import { TextCell } from "component/common/Table/cells/TextCell/TextCell";
+import PermissionSwitch from "component/common/PermissionSwitch/PermissionSwitch";
+import { useRequiredPathParam } from "hooks/useRequiredPathParam";
+import { Dialogue } from "component/common/Dialogue/Dialogue";
+import { ConditionallyRender } from "component/common/ConditionallyRender/ConditionallyRender";
+import { useChangeRequestConfig } from "../../../../../hooks/api/getters/useChangeRequestConfig/useChangeRequestConfig";
+import { useChangeRequestApi } from "../../../../../hooks/api/actions/useChangeRequestApi/useChangeRequestApi";
+import { UPDATE_PROJECT } from "@server/types/permissions";
 
 export const ChangeRequestConfiguration: VFC = () => {
     const [dialogState, setDialogState] = useState<{
         isOpen: boolean;
         enableEnvironment?: string;
+        isEnabled: boolean
     }>({
         isOpen: false,
         enableEnvironment: '',
+        isEnabled: false
     });
     const projectId = useRequiredPathParam('projectId');
-    const data = [
-        {
-            environment: 'dev',
-            type: 'test',
-            isEnabled: false,
-        },
-    ] as any[]; // FIXME: type
+    const { data, loading, refetchChangeRequestConfig } = useChangeRequestConfig(projectId);
+    const { updateChangeRequestEnvironmentConfig } = useChangeRequestApi()
 
-    const onClick = (enableEnvironment: string) => () => {
-        setDialogState({ isOpen: true, enableEnvironment });
+    const onClick = (enableEnvironment: string, isEnabled: boolean) => () => {
+        setDialogState({ isOpen: true, enableEnvironment, isEnabled });
     };
 
+    const onConfirm = async () => {
+        if (dialogState.enableEnvironment) {
+            await updateChangeRequestEnvironmentConfig(projectId, dialogState.enableEnvironment, !dialogState.isEnabled);
```

Comment:
This needs to be try catched, and display a toast error if we don't succeed.

Reply:
Done